src/prefect/agent/docker/agent.py

```diff
@@ -458,25 +458,29 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
             # would be empty after cleaning
             or flow_run.id
         )
-        # Checking for container names here is a slight deploy performance hit. If
-        # it ends up being an issue we can move to a try/except model
-        existing_names = {c["name"] for c in self.docker_client.containers()}
-        index = 0
-        while container_name in existing_names:
-            index += 1
-            container_name = f"{slugified_name}-{index}"
-
-        # Create the container
-        container = self.docker_client.create_container(
-            image,
-            command=get_flow_run_command(flow_run),
-            environment=env_vars,
-            name=container_name,
-            volumes=container_mount_paths,
-            host_config=self.docker_client.create_host_config(**host_config),
-            networking_config=networking_config,
-            labels=labels,
-        )
+
+        # Create the container with retries on name conflicts
+        index = 0  # will be bumped on name colissions
+        while True:
+            try:
+                container = self.docker_client.create_container(
+                    image,
+                    command=get_flow_run_command(flow_run),
+                    environment=env_vars,
+                    name=container_name,
+                    volumes=container_mount_paths,
+                    host_config=self.docker_client.create_host_config(**host_config),
+                    networking_config=networking_config,
+                    labels=labels,
+                )
+            except docker.errors.APIError as exc:
+                if "Conflict" in str(exc) and "container name" in str(exc):
+                    index += 1
```

Comment:
Is this necessary? Seems like it could be simpler to just append a bit of a UUID if a conflict is found?

Reply:
I guess this is guaranteed to work eventually whereas a bit of UUID isn't actually universally unique and we'd still have to loop. Easier to test this way too. We can revisit the implementation in the future, this has been a nightmare to try to fix quickly :(